server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1072,6 +1075,7 @@ private void doDeleteShardSnapshots(
                     repositoryStateId,
```

Comment:
had conflict in this method during backport due to `useShardGenerations` check in 2.x